File: 프로그래머스/Lv.1/12947. 하샤드 수/하샤드 수.cs
public class Solution {
    public bool solution(int x) {
        bool answer = true;
        int sum = 0;
        int dev =1 ;
        int result;
        while(x/dev != 0)
        {
            result = (x/dev)%10;
            sum += result;
            dev *= 10;
        }
        if(x % sum != 0)
            answer = false;
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        long denominator = 0;
        long number = x;
        while(number > 0)
        {
            denominator += number % 10;
            number /= 10;
        }
        return answer = x % denominator == 0 ? true : false;
    }
}